src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -118,6 +118,15 @@ DxFontRenderData::DxFontRenderData(::Microsoft::WRL::ComPtr<IDWriteFactory1> dwr
                                                                                                   DWRITE_FONT_STYLE style,
                                                                                                   DWRITE_FONT_STRETCH stretch)
 {
+    const auto defaultIt = _textFormatMap.find(_defaultFontInfo);
+    if (weight == _defaultFontInfo.GetWeight() &&
+        style == _defaultFontInfo.GetStyle() &&
+        stretch == _defaultFontInfo.GetStretch() &&
+        defaultIt != _textFormatMap.end())
+    {
+        return (*defaultIt).second;
+    }

```

Comment:
Since we only use weight, style and stretch for hashing now, this lookup is redundant with the existing lookup below.

Also I'd suggest to not use `DxFontInfo` at all, but rather introduce a small struct that contains just the necessary 3 fields as a key for the `std::unordered_map`.

Reply:
We can keep the shortcut comparison with `_defaultFontInfo` though and skip the hashing part.

Reply:
So, if we skip the iterator, end and second bits and just return `_defaultFontInfo` directly? Seems reasonable to me :smile:

Reply:
The default font should be the absolute most common case in my opinion. It might be worthwhile to add such a shortcut as it's just 2 lines of extra code.